test/jdk/tools/sincechecker/SinceChecker.java

```diff
@@ -0,0 +1,769 @@
+
+note on real vs effective `@since`:
+
+Real since value of an API element is computed as the oldest release in which the given API element was introduced. That is:
+- for modules, classes and interfaces, the release in which the element with the given qualified name was introduced
+- for constructors, the release in which the constructor with the given VM descriptor was introduced
+- for methods and fields, the release in which the given method or field with the given VM descriptor became a member of its enclosing class or interface, whether direct or inherited
+
+Effective since value of an API element is computed as follows:
+- if the given element has a `@since` tag in its javadoc, it is used
+- in all other cases, return the effective since value of the enclosing element
+
+Special Handling for preview method:
+- When an element is still marked as preview, the `@since` should be the first JDK release where the element was added.
+- If the element is no longer marked as preview, the `@since` should be the first JDK release where it was no longer preview.
+
+note: The `<unique-Element-ID>` for methods looks like `method: <erased-return-descriptor> <binary-name-of-enclosing-class>.<method-name>(<ParameterDescriptor>)`.
+it is somewhat inspired from the VM Method Descriptors. But we use the erased return so that methods that were later generified remain the same.
+*/
+
+
+public class SinceChecker {
+    private final Map<String, Set<String>> LEGACY_PREVIEW_METHODS = new HashMap<>();
+    private final Map<String, IntroducedIn> classDictionary = new HashMap<>();
+    private final JavaCompiler tool;
+    private final List<String> wrongTagsList = new ArrayList<>();
+
+    public static void main(String[] args) throws Exception {
+        if (args.length == 0) {
+            throw new SkippedException("Test module not specified");
+        }
+        SinceChecker sinceCheckerTestHelper = new SinceChecker();
+        sinceCheckerTestHelper.testThisModule(args[0]);
+    }
+
+    private SinceChecker() throws IOException {
+        tool = ToolProvider.getSystemJavaCompiler();
+        for (int i = 9; i <= Runtime.version().feature(); i++) {
+            JavacTask ct = (JavacTask) tool.getTask(null, null, null,
+                    List.of("--release", String.valueOf(i)),
+                    null,
+                    Collections.singletonList(SimpleJavaFileObject.forSource(URI.create("myfo:/Test.java"), "")));
+            ct.analyze();
+
+            String version = String.valueOf(i);
+            Elements elements = ct.getElements();
+            elements.getModuleElement("java.base"); // forces module graph to be instantiated
+            elements.getAllModuleElements().forEach(me ->
+                    processModuleRecord(me, version, ct));
+        }
+    }
+
+    private void processModuleRecord(ModuleElement moduleElement, String releaseVersion, JavacTask ct) {
```

Comment:
for this method, and similarly named methods, the use of `Record` seems confusing, if only because there do not seem to be any record classes being processed or analyzed.   Consider dropping `Record` or changing it to `Element`. Same for `analyzePackageRecord`, `analyzeClassRecord`.

Also, consider being consistent with `process` or analyze`.

Reply:
Fixed in 32edb4d811978f87bd2f850c214191e8fc7b8c88